src/main/java/org/dependencytrack/tasks/OSVDownloadTask.java

```diff
@@ -0,0 +1,174 @@
+                    ZipInputStream zipIn = new ZipInputStream(url.openStream());
+                    unzipFolder(zipIn);
+                    zipIn.closeEntry();
+                }
+            } catch (IOException exception) {
+                exception.printStackTrace();
+            }
+        }
+    }
+
+    private void unzipFolder(ZipInputStream zipIn) throws IOException {
+
+        BufferedReader reader;
+        GoogleOSVAdvisoryParser parser = new GoogleOSVAdvisoryParser();
+        try {
+            ZipEntry zipEntry = zipIn.getNextEntry();
+            while (zipEntry != null) {
+
+                reader = new BufferedReader(new InputStreamReader(zipIn));
+                String line = null;
+                StringBuilder out = new StringBuilder();
+                while ((line = reader.readLine()) != null) {
+                    out.append(line);
+                }
+                JSONObject json = new JSONObject(out.toString());
+                System.out.println(json);
+                final OSVAdvisory osvAdvisory = parser.parse(json);
+                updateDatasource(osvAdvisory);
+                zipEntry = zipIn.getNextEntry();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void updateDatasource(final OSVAdvisory advisory) {
+        LOGGER.info("Updating datasource with Google OSV advisories");
+        try (QueryManager qm = new QueryManager()) {
+
+            LOGGER.debug("Synchronizing Google OSV advisory: " + advisory.getId());
+            final Vulnerability synchronizedVulnerability = qm.synchronizeVulnerability(mapAdvisoryToVulnerability(qm, advisory), false);
+            final List<VulnerableSoftware> vsList = new ArrayList<>();
+            for (OSVVulnerability osvVulnerability: advisory.getVulnerabilities()) {
+                VulnerableSoftware vs = mapVulnerabilityToVulnerableSoftware(qm, osvVulnerability);
+                if (vs != null) {
+                    vsList.add(vs);
+                }
+            }
+            LOGGER.debug("Updating vulnerable software for OSV advisory: " + advisory.getId());
+            qm.persist(vsList);
+            synchronizedVulnerability.setVulnerableSoftware(vsList);
+            qm.persist(synchronizedVulnerability);
+        }
+        Event.dispatch(new IndexEvent(IndexEvent.Action.COMMIT, Vulnerability.class));
+    }
+
+    private Vulnerability mapAdvisoryToVulnerability(final QueryManager qm, final OSVAdvisory advisory) {
+
+        final Vulnerability vuln = new Vulnerability();
+        vuln.setSource(Vulnerability.Source.GOOGLE);
+        vuln.setVulnId(String.valueOf(advisory.getId()));
+        vuln.setTitle(advisory.getSummary());
+        vuln.setDescription(advisory.getDetails());
+        vuln.setPublished(Date.from(advisory.getPublished().toInstant()));
+        vuln.setUpdated(Date.from(advisory.getModified().toInstant()));
+
```

Comment:
There are some more fields we could ingest, e.g. the CVSSv3 vector and credits.

Reply:
What would be credits here? OSV have credits with name and contact https://ossf.github.io/osv-schema/#credits-fields

Reply:
A comma-separated list of names should suffice. Using credits is currently only being done for VulnDB, see

https://github.com/DependencyTrack/dependency-track/blob/a3b42a6fe4d19fc91122887dba0841f26fa7c7c4/src/main/java/org/dependencytrack/parser/vulndb/ModelConverter.java#L113-L131

I don't think we need the contact information and can safely discard it.